server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -91,6 +142,38 @@ static ClusterState modifyDataStream(
         return ClusterState.builder(currentState).metadata(updatedMetadata).build();
     }
 
+    /**
+     * Creates an updated cluster state in which the requested data streams have the data lifecycle provided.
+     * Visible for testing.
+     */
+    static ClusterState modifyDataLifecycle(
```

Comment:
Might be a personal preference but should we name this `updateDataLifecycle` ?

Reply:
No strong feelings here, I will change it.